deepspeed/pt/deepspeed_config_utils.py

```diff
@@ -8,18 +8,12 @@
 
 
 def get_scalar_param(param_dict, param_name, param_default_value):
-    if param_name in param_dict.keys():
-        return param_dict[param_name]
-    else:
-        return param_default_value
+    return param_dict.get(param_name, param_default_value)
 
 
 def dict_raise_error_on_duplicate_keys(ordered_pairs):
     """Reject duplicate keys."""
-    d = {}
-    for k, v in ordered_pairs:
-        if k in d:
-            raise ValueError("Duplicate key in DeepSpeed config: %r" % (k, ))
-        else:
-            d[k] = v
+    d = dict((k, v) for k, v in ordered_pairs)
+    if len(d) != len(ordered_pairs):
+        raise ValueError("Duplicate key in DeepSpeed config: %r" % (k, ))

```

Comment:
I believe the error message needs to be updated after this change. `k` will no longer contain the duplicate key.

Reply:
Oops. I missed that.